R/data.table.R

```diff
@@ -1027,10 +1027,14 @@ replace_dot_alias = function(e) {
           lhs = jsub[[2L]]
           jsub = jsub[[3L]]
           if (is.name(lhs)) {
-            lhs = as.character(lhs)
+            if (deparse(lhs) == '.SD') lhs = sdvars else lhs = as.character(lhs)
```

Comment:
checking against `as.name(".SD")` should be preferred rather than `deparse` which has more corner cases to handle